core/src/main/java/com/oath/oak/OakUnsafeDirectBuffer.java

```diff
@@ -0,0 +1,59 @@
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+/**
+ * This interface allows high performance access to the underlying data of Oak.
+ * To achieve that, it sacrifices safety, so it should be used only if you know what you are doing.
+ * Misuse of this interface might result in corrupted data, a crash or a deadlock.
+ *
+ * Specifically, the developer should be concerned by two issues:
+ * 1. Concurrency: using this interface inside the context of serialize, compute, compare and transform is thread safe.
+ *    In other contexts (e.g., zero copy API), the developer should ensure that there is no concurrent access to this
+ *    data. Failing to ensure that might result in corrupted data.
+ * 2. Data boundaries: when using this interface, Oak will not alert the developer regarding any out of boundary access.
+ *    Thus, the developer should use getOffset() and getLength() to obtain the data boundaries and carefully access
+ *    the data. Writing data out of these boundaries might result in corrupted data, a crash or a deadlock.
+ *
+ * To use this interface, the developer should cast Oak's buffer (OakRBuffer/OakWBuffer) to this interface,
+ * similarly to how Java's internal DirectBuffer is used. For example:
+ * void foo(OakRBuffer b) {
+ *     long address = ((OakUnsafeDirectBuffer) b).getAddress();
+ *     ...
+ * }
```

Comment:
Nice documentation.
When all PRs are resolved this should also go into the wiki, with code examples of using the interface, e.g., the way you are using it in Druid

Reply:
Please add also about read-only and writable buffers when cast OakRBuffer or OakWBuffer respectively

Reply:
I see that you speak about it bellow, but I think it should be correlated with cast...

Reply:
fixed